src/utilities/EventEmitter.ts

```diff
@@ -231,96 +238,60 @@ export default class EventEmitter {
 
   /**
    * Listen for a single occurrence of an event
-   * @param event the name of the event to listen to
+   * @param listenerOrEvents (optional) the name of the event to listen to
+   * @param listener (optional) the listener to be called
    */
-  once(event: string): Promise<void>;
-
-  /**
-   * Listen for a single occurrence of any event
-   * @param listener the listener to be called
-   */
-  once(listener: Function): void;
-
-  /**
-   * Listen for a single occurrence of an event
-   * @param event the name of the event to listen to
-   * @param listener the listener to be called
-   */
-  once(event?: string | string[] | null, listener?: Function): void;
-
-  once(...args: unknown[]): void | Promise<void> {
-    const argCount = args.length;
-    if ((argCount === 0 || (argCount === 1 && typeof args[0] !== 'function')) && Promise) {
-      const event = args[0];
+  once<K extends EventKey<T>>(
+    listenerOrEvents?: K | K[] | EventListener<T[K]>,
+    listener?: EventListener<T[K]>
+  ): void | Promise<any> {
+    // .once()
+    if ([listenerOrEvents, listener].filter((i) => i).length === 0) {
```

Comment:
I don't think it's immediately obvious what this line is doing with the filter. I would add a comment explaining it or opt a more verbose but self-explanatory way of doing this

Reply:
I agree, although the pattern `.filter((i) => !!i)` is common in much JS I think `if(!(listenerOrEvents || listener))` is clearer here or ```const argumentExists = listenerOrEvents || listener;
if(!argumentExists)...```

Reply:
I would think this is fairly idiomatic, but perhaps that's subjective. We can also use the more old school `arguments.lenght > 0`.